interface/slack.py

```diff
@@ -45,14 +44,33 @@ def get_channel_users(self, channel_id):
             "conversation.members",
             channel=channel_id
         )
-        if 'ok' not in response:
+        if 'ok' not in response or response['ok'] is False:
             logging.error("User retrieval "
                           "from channel {} failed with error: {}".
                           format(channel_id, response['error']))
             raise SlackAPIError(response['error'])
         else:
-            json_response = json.loads(response)
-            return json_response["members"]
+            return response["members"]
+
+    def create_channel(self, channel_name):
+        """
+        Create a channel with the given name.
+
+        :return name of newly created channel
+        """
+        logging.debug("Attempting to create channel with name {}".
+                      format(channel_name))
+        response = self.sc.api_call(
+            "channels.create",
+            name=channel_name,
+            validate=True
+        )
+        if 'ok' not in response or response['ok'] is False:
+            logging.error("Channel creation "
+                          "with name {} failed with error: {}".
+                          format(channel_name, response['error']))
```

Comment:
So my thought here is that, if this fails for a reason other than `name_taken`, it would still be a good idea to raise an exception and let the caller handle it (which is the behaviour of all the other calls) - though in this case maybe the caller just catches and ignores the exception. So I think it would be best to add a new case to the `if` statement that checks for `name_taken` and in that case logs a warning (my personal preference would be to send to `logging.warn` instead of `logging.error`, but logging levels don't really matter that much 😛) and returns, and keep this case the same as it was before (i.e. raising an exception).

Reply:
Hmmm I'm a little bit confused, I will ask tomorrow ahaha